### PR TITLE
Remove old custom op implementation

### PR DESCRIPTION
**Pull request by smessmer**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#21085 Remove old custom op implementation**&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D15542261/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #21084 Add aliasAnalysis to torch::RegisterOperators()&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D15542097/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #21181 c10::OperatorOptions&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D15569897/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #21079 registration options should only be callable on rvalues&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D15541583/)

Now that torch::jit::RegisterOperators() always passes through to torch::RegisterOperators() (see diffs stacked below this), we can remove the old custom op implementation.

Differential Revision: [D15542261](https://our.internmc.facebook.com/intern/diff/D15542261/)

### CODE REVIEW

File: torch/csrc/jit/custom_operator.h [hunk 1]
```diff
@@ -3,218 +3,9 @@
...
-  //   2. Have some way of distinguishing between a custom op and a builtin op
-  //      so that we can apply the above rule.
-  // We do this by manually whitelisting "aten" and "prim" namespaces as
-  // builtins.
-  //
-  // We don't want to preserve this distinction between custom/builtin ops, as
-  // it is fragile and hard to maintain. When we provide a way for op
-  // registration to specify alias annotations, we should fix up builtins to
-  // use that and remove all references to this note.
-  Symbol name = Symbol::fromQualString(schema.name());
```
Comment (dzhulgakov) [inline torch/csrc/jit/custom_operator.h]: that was kind of a good check, but I'm not sure where to plug it in the new version. Is there a public only API wrapper?

File: test/cpp/jit/test_custom_operators.h [hunk 2]
```diff
@@ -138,94 +136,14 @@ void testCustomOperators() {
...
-    ASSERT_THROWS_WITH(
-        createOperator(
-            "aten::op(float[] f) -> int",
-            [](const std::vector<double>& f) -> int64_t { return f.size(); }),
-        "Tried to register a custom operator to a reserved namespace");
-  }
 }
 
 void testCustomOperatorAliasing() {
-  RegisterOperators reg({createOperator(
```
Comment (dzhulgakov) [inline test/cpp/jit/test_custom_operators.h]: so we didn't consider `createOperator` as part of public API, did we? (tutorial doesn't refer to it, so should be ok)

**Review by dzhulgakov — APPROVED** [review]
Seems good

**Review by zdevito — APPROVED** [review]

File: torch/csrc/jit/custom_operator.h (hunk 1 above)
Comment (smessmer) [reply torch/csrc/jit/custom_operator.h]: no, we intentionally use the same API internally and externally. Maybe it's time to actually do what the comment is suggesting.

File: test/cpp/jit/test_custom_operators.h (hunk 2 above)
Comment (smessmer) [reply test/cpp/jit/test_custom_operators.h]: not that I'm aware of. @zdevito ?

**Pull request closed without merging**